src/Polly.Core/ResiliencePipelineBuilderBase.cs

```diff
@@ -55,7 +55,19 @@ private protected ResiliencePipelineBuilderBase(ResiliencePipelineBuilderBase ot
     public string? InstanceName { get; set; }
 
     /// <summary>
-    /// Gets or sets a <see cref="TimeProvider"/> that is used by strategies that work with time.
+    /// Gets or sets the <see cref="Polly.ResilienceContextPool"/> associated with the builder.
+    /// </summary>
+    /// <remarks>
+    /// A custom pool can be used to configure custom behavior for  creation.
+    /// This can include setting a default <c>continueOnCapturedContext</c> parameter or custom operation key resolution.
+    /// </remarks>
+    /// <value>
+    /// If the default value of <see langword="null"/> is used, <see cref="ResilienceContextPool.Shared"/> will be used.
+    /// </value>
+    public ResilienceContextPool? ResilienceContextPool { get; set; }
```

Comment:
Nit: keep it as now or just shorten to `ContextPool`. I'll leave it up to you.

Reply:
I like `ContextPool` better as well, changing!